addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -308,12 +308,28 @@ var Wysiwyg = Widget.extend({
             await this.editor.execCommand(onSaveLinkDialog);
         });
     },
-    openMediaDialog() {
-        var mediaDialog = new weWidgets.MediaDialog(this,
-            {},
-        );
+    openMediaDialog(params) {
+        let mediaDialogOptions = {}
+        let htmlClass = ""
+
+        // Customize the look and behavior of the media modal when
+        // the origin of the media modal is a FA icon double click
+        if (params.origin === "fontAwesomeDoubleClick") {
+            mediaDialogOptions.noImages = true;
+            mediaDialogOptions.noDocuments = true;
+            mediaDialogOptions.noVideos = true;
+
+            const faZoomClassRegex = RegExp('fa-[0-9]x');
+            function filterUnwantedClasses(className) {
+                return !className.startsWith('fa') || faZoomClassRegex.test(className);
+            }
+            htmlClass = [...params.htmlClass].filter(filterUnwantedClasses).join(" ")
```

Comment:
why do we need this?

Reply:
We need this to be able to keep the style of a icon we just replace. otherwise we loose the size, color etc of the existing icon.
But a the same time we cannot keep the old fa class, or the icon would not be accurate.